File: src/main/java/com/github/nsc/de/compiler/parser/node/variables/VariableAssignmentNode.java
package com.github.nsc.de.compiler.parser.node.variables;

import com.github.nsc.de.compiler.parser.node.Node;
import com.github.nsc.de.compiler.parser.node.ValuedNode;

public class VariableAssignmentNode implements ValuedNode {

    private final String name;
    private final Node value;

    public VariableAssignmentNode(String name, Node value) {
        this.name = name;
        this.value = value;
    }

    public String getName() {
        return name;
    }

    public Node getValue() {
        return value;
    }

    @Override
    public String toString() {
        return this.getName() + " = " + this.getValue();
    }
}

File: parser/src/main/kotlin/com/github/nsc/de/shake/parser/node/variables/VariableAssignmentNode.kt
package com.github.nsc.de.shake.parser.node.variables

import com.github.nsc.de.shake.parser.node.*
import com.github.nsc.de.shake.util.characterinput.position.PositionMap

class VariableAssignmentNode(map: PositionMap, val variable: ValuedNode, val value: Node, val operatorPosition: Int) :
    ValuedNode(map) {

    override fun toJson(): Map<String, *> = mapOf(
        "name" to "VariableAssignmentNode",
        "variable" to variable.json,
        "value" to value.json
    )
}